.github/workflows/lib-lint-and-test.yaml

```diff
@@ -66,21 +66,28 @@ jobs:
         uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
         with:
           persist-credentials: false
-      - name: Set up Python
-        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
+
+      - name: Install uv and set the python version
+        uses: astral-sh/setup-uv@d0cc045d04ccac9d8b7881df0226f9e82c39688e # v6.8.0
         with:
+          enable-cache: false
           python-version: "3.12"
-      - name: Install tox
+          version: "0.8.15"
+
+      - name: Installing dependencies
         working-directory: library
-        run: |
-          pip install '.[dev]'
-      - name: Code quality checks
+        run: uv sync --locked --extra dev --extra cuda --no-extra xpu
```

Comment:
Is the extra `cuda` necessary for the tests to pass? Does it even work on the standard GPU-less Github runners?